Searching messages in a room (not encrypted)

alanpoon opened the pull request:
![Unknown](https://github.com/user-attachments/assets/2b064196-9b1c-4a29-8493-132c9759b73b)



https://github.com/user-attachments/assets/97831448-af17-46ac-97b2-2adf171d232a

alanpoon committed:
added highlight

alanpoon committed:
added search message logic

alanpoon committed:
Merge branch 'main' into search_messages#122

alanpoon committed:
resolve conflict

alanpoon committed:
add search_all_room

alanpoon committed:
added room_name in search result

alanpoon committed:
added small_state display in search result

alanpoon committed:
EventableWrapperETI

alanpoon committed:
before populate_small_state

alanpoon committed:
migrate to use generic

alanpoon committed:
Added early abort search message

alanpoon committed:
t

alanpoon committed:
t

alanpoon committed:
added body_inner

alanpoon committed:
fix condensed message in search result

alanpoon committed:
used relations

alanpoon committed:
Merge remote-tracking branch 'origin/main' into seacr2

alanpoon committed:
rebase

alanpoon committed:
Merge branch 'main' into search_messages_3#122

aaravlu commented on src/sliding_sync.rs:
```diff
@@ -381,6 +382,17 @@ pub enum MatrixRequest {
         matrix_id: MatrixId,
         via: Vec<OwnedServerName>
     },
+    /// General Matrix Search API with given categorie
```
Agree.

alanpoon committed:
move room_search_result underneath room_screen mod

alanpoon committed:
Merge branch 'main' into search_messages_3#122

smarizvi110 commented:
https://github.com/user-attachments/assets/be783d62-ecd3-4de1-a7e7-f92894cab9f8

Hi @alanpoon! This PR looks wonderful! I checked it out after your request to do so on #395 and found a few issues. The attached video walks through them.

Firstly, the initial start up sequence looks very different from what the program looks like when cloned from the original repo's main branch. Secondly, several icons seem to be missing too. I checked out the commits ahead of your branch on `project-robius/robrix:main` and it doesn't seem like there was any file changed that would lead to this issue, but I could be wrong on this.

Third, the search feature itself seems to work! The only problem is that when you attempt to **search all rooms** while initiating a search in an encrypted room, it does not seem to search in any available unencrypted rooms either and simply tells me that searching unencrypted rooms is currently not supported. Furthermore, the little notification that tells me this seems to _never_ go away, and just stays there in the top right corner. Perhaps that should be tweaked to disappear in a few seconds?

One last thing is that the mermaid diagram you attached to the PR here makes sense, but I don't quite understand why there are 4 edges coming out of the `Check Room Encryption` node. Shouldn't there just be 2, one for encrypted room searching flow and one for unencrypted room search flow? If I'm misunderstanding something, please do let me know!

Thank you!

alanpoon committed:
search_button_wrap improvement

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon commented:
> MUTED_2025-05-22.18-27-24.mp4 
> 
> Hi @alanpoon! This PR looks wonderful! I checked it out after your request to do so on #395 and found a few issues. The attached video walks through them (please skip to 0:19)
> 
> Firstly, the initial start up sequence looks very different from what the program looks like when cloned from the original repo's main branch. Secondly, several icons seem to be missing too. I checked out the commits ahead of your branch on `project-robius/robrix:main` and it doesn't seem like there was any file changed that would lead to this issue, but I could be wrong on this.
> 
> Third, the search feature itself seems to work! The only problem is that when you attempt to **search all rooms** while initiating a search in an encrypted room, it does not seem to search in any available unencrypted rooms either and simply tells me that searching unencrypted rooms is currently not supported. Furthermore, the little notification that tells me this seems to _never_ go away, and just stays there in the top right corner. Perhaps that should be tweaked to disappear in a few seconds?
> 
> One last thing is that the mermaid diagram you attached to the PR here makes sense, but I don't quite understand why there are 4 edges coming out of the `Check Room Encryption` node. Shouldn't there just be 2, one for encrypted room searching flow and one for unencrypted room search flow? If I'm misunderstanding something, please do let me know!
> 
> For reference, the second video shows what Robrix looks like currently on `project-robius/robrix:main` where the icons are not missing and the startup screen looks a little more user-friendly.
> 
> Please note that in both cases, I gave Robrix my credentials through the CLI when using `cargo run`.
> 
>  2025-05-22.18-30-28.mp4 
> 
> Thank you!

Thank you for testing.
Unfortunately, I cannot replicate the missing icon issues even after cloning my branch. I have fixed the issue of not be able to search all rooms while initiating a search in an encrypted room. There's supposed to be a close button on  the top right of the popup which could be missing due to missing icon issues. 
I have also fixed the mermaid diagram.

alanpoon committed:
abort previous search request

alanpoon committed:
allow finding messages in encrypted room

alanpoon committed:
Merge branch 'search_messages_3#122' of https://github.com/alanpoon/robrix into search_messages_3#122

alanpoon committed:
fix selection

alanpoon committed:
fix clippy

alanpoon committed:
fix spelling

kevinaboos commented on src/sliding_sync.rs:
```diff
@@ -1102,6 +1117,139 @@ async fn async_worker(
+                                        })
+                                    });
+                                }
+                                item.context.events_after.iter().rev().for_each(|f| {
+                                    if let Ok(timeline_event) = f.deserialize() {
+                                        items
+                                            .push(SearchResultItem::ContextEvent(timeline_event));
+                                    }
+                                });
+                                let timestamp = match event {
+                                    AnyTimelineEvent::MessageLike(ref event) => {
+                                        if let Some(replace) = event.relations().replace {
+                                            replace.origin_server_ts()
+                                        } else {
+                                            event.origin_server_ts()
+                                        }
+                                    }
+                                    _ => event.origin_server_ts(),
+                                };
+                                let room_id = event.room_id().to_owned();
+                                items.push(SearchResultItem::Event(event));
+
+                                item.context.events_before.iter().rev().for_each(|f| {
```
please don't use `for_each()`, especially not for long multi-line closures. It should really only be used for a tiny one-liner. I've mentioned this before, but to reiterate, the reason for this is that it's easy to make the mistake of trying to return from the parent function within the closure, which won't work. I just observed this exact bug occur in Makepad itself, so in summary, I'd consider usage of `for_each` to be an anti-pattern (except for _very_ short one-line closures).

kevinaboos commented on src/home/room_screen.rs:
```diff
@@ -3959,7 +4337,7 @@ fn populate_small_state_event(
 /// Sets the text of the `Label` at the given `item`'s live ID path
 /// to a typical 12-hour AM/PM timestamp format.
-fn set_timestamp(
+pub fn set_timestamp(
```
I just removed this `set_timestamp()` function in favor of a dedicated `Timestamp` widget. You'll need to pull down and merge in the latest changes from `main` to resolve this conflict.

kevinaboos commented on src/home/new_message_context_menu.rs:
```diff
@@ -620,3 +620,9 @@ impl NewMessageContextMenuRef {
         inner.show(cx, details)
     }
 }
+pub trait ContextMenuFromEvent {
```
why do we need modifications for the context menu here? What is the relationship between the message context menu and the search results? Seems like they're completely unrelated.

kevinaboos reviewed (changes requested):
Thanks Alan, great start on this issue!

Personally, I find the way that Element displays contextual messages surrounding a search message to be very confusing. Let's just... not do that, at all.
As I said in #122, we should design a search UI similar to how Discord does it. Kindly refer to that issue and its comments for more details.

-------------------

I appreciate how you tried to re-use the RoomScreen widget to display the search results, however I don't think that's the proper approach in this case. We need to create a completely new widget for displaying search results (which could share/reuse some of the components from RoomScreen, if you want).

The reason for this is that we'll want to add in completely new UI elements that don't belong in a regular RoomScreen. For example, when searching all rooms, it is hard to tell which messages come from which rooms. Instead of using a RoomHeader with just the room name acting as a divider, we should show a small view with the room name + avatar (and maybe the date) on the left of the messages from that room. This will make it a lot clearer where a message comes from, and is not possible to do by re-using the RoomScreen widget to display search results. 

Another reason why we need a new widget for showing search results is because we want to be able to show the search results in a separate dock tab than the original room. Again, see how Discord shows the results in a side pane, which allows the user to continue viewing the current room without losing their place.

<img width="700" alt="image" src="https://github.com/user-attachments/assets/a9a38732-ab38-4151-b334-9847cf8395fb" />

------------
This PR introduces far too much complexity, with the addition of all of these traits, e.g., `MessageDisplay` and all of its wrapper structs.

I assume this complexity is there in order to allow you to re-use the RoomScreen widget to show search results, so I assume that you can remove these traits and complicating factors once you start using a dedicated widget for the search results.

Also, some Rust advice: in general, when you add a new trait, you do not need to introduce wrapper types for it, since you can implement your new trait for foreign types without issue. The wrapper type pattern is useful when you want to implement a _foreign_ trait for a _foreign_ type, but you don't need that here.

--------------

I tested this PR on my mac and most stuff seems to work, so nice job! There are a few pending problems/adjustments that i'd like you to address.
* Remove the surrounding contextual messages, as mentioned above.
* Lots of messages just don't load at all. The reply previews are all blank, and many regular messages also are completely blank. Not sure why.
   *  <img width="700" alt="image" src="https://github.com/user-attachments/assets/571cb58d-aa74-4a61-bade-d95c1cc268db" />
* Highlighting the search term is a nice touch, i like that! But the light green color is not obvious -- let's use a "highlighter yellow" color, e.g., something around 0xFFFF00.
* The search input bar causes the dock tab bar to be placed beneath the search bar, which looks weird. I'm not sure if that was intentional, but it doesn't look right. 
  * If you intended to fix it but don't know how, kindly let me know next time with a comment on the PR before I start my review, which will draw attention to it.
  * If it's still a pending change that you haven't made yet, then that's fine, but then I'd ask you to kindly not request a review from me until everything is ready. 
  * If you're looking for design suggestions, I would vote that we put the message search bar either above the "All Rooms" label. In the future we can have a single integrated input bar for both room filters and message search, but that requires extra functionality from Makepad, so we'll save that for later.

alanpoon committed:
remove specific ruma

alanpoon committed:
don't use trait

alanpoon committed:
separate populate_message_search_view

alanpoon committed:
display search result on the right

alanpoon committed:
highlight yellow

alanpoon committed:
rounded view for search result

alanpoon committed:
revert roomscreen

alanpoon committed:
cleanup

alanpoon committed:
added date for search result

alanpoon committed:
Change to search_screen to right_panel

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
right

alanpoon committed:
click search input to open

alanpoon committed:
search

alanpoon committed:
added right panel

alanpoon committed:
remove cancel_button

alanpoon committed:
Fix

alanpoon committed:
added doc

alanpoon committed:
cargo fmt

alanpoon committed:
remove unneccessary changes

alanpoon committed:
minor code improvement

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -292,72 +299,85 @@ live_design! {
+                width: Fill, height: Fill,
+                show_bg: true,
+                visible: true,
+                draw_bg: {color: (#FFFFFF00),}
```
Fix

alanpoon committed:
SearchResultView fix

alanpoon committed:
Merge branch 'main' into search_messages_4#122

alanpoon committed:
Save Previous search result

alanpoon committed:
Added scroll to message in roomscreen

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
code restructure

alanpoon committed:
added constraint to search term length

alanpoon committed:
string optimization

alanpoon committed:
update cargo lock

alanpoon committed:
revert cargo lock

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
search again button

alanpoon committed:
Added search input limit as const

alanpoon committed:
speed up Scroll To Message from search result

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
remove set_room_screen_widget_uid

alanpoon committed:
remove populate_message_search_view's search widget_uid

kevinaboos commented on src/sliding_sync.rs:
```diff
@@ -1147,6 +1155,156 @@ async fn async_worker(
+                    continue;
+                }
+                
+                // Validate search term length
+                if search_term.len() < MIN_SEARCH_TERM_LENGTH {
+                    log!("Search term too short: '{}' (minimum {} characters)", search_term, MIN_SEARCH_TERM_LENGTH);
+                    enqueue_popup_notification(PopupItem {
+                        message: format!("Search term must be at least {} characters long", MIN_SEARCH_TERM_LENGTH),
+                        auto_dismissal_duration: Some(3.0),
+                        kind: PopupKind::Info,
+                    });
+                    continue;
+                }
```
why restrict the user like this?

If they really want to search for something vague like "a", why would we stop them? I don't understand the purpose of all of these unnecessary limitations. Please only limit things when it is required.

alanpoon replied:
removed

kevinaboos commented on src/sliding_sync.rs:
```diff
@@ -1147,6 +1155,156 @@ async fn async_worker(
+                                    }
+                                }
+                                items.push_back(right_panel::search_message::SearchResultItem::Event{ event:
+                                    Box::new(event), formatted_content: Box::new(message_option)}
```
why are we boxing up these two states?

alanpoon replied:
Change to using Arc

kevinaboos commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+    if let MessageType::Text(text) = &mut message.msgtype {
+        let formatted = if let Some(ref mut formatted) = text.formatted {
+            let mut body = formatted.body.clone();
+            // TODO: Remove all <code> </code> before appending them to highlights.
+            for highlight in highlights {
+                body = body.replace(highlight, &format!("<code>{}</code>", highlight));
+            }
+            body
+        } else {
+            let mut formatted_string = text.body.clone();
+            for highlight in highlights {
+                formatted_string =
+                    formatted_string.replace(highlight, &format!("<code>{}</code>", highlight));
```
why would you want to use `<code>` tags for this? If you just want to highlight it, then use the `font` tag or `span` tag to set the text color and/or the background color.

The attributes that you need to use are described in the `MatrixHtmlSpan` struct: https://github.com/project-robius/robrix/blob/d089e7e2a29e350f19b9e2ac3812ee4110ff8eac/src/shared/html_or_plaintext.rs#L451-L454

alanpoon replied:
Fixed

kevinaboos commented on src/sliding_sync.rs:
```diff
@@ -1147,6 +1155,156 @@ async fn async_worker(
                 });
             }
+            MatrixRequest::SearchMessages {
+                room_id,
+                search_term,
+                include_all_rooms,
```
I think this request should just directly accept the `v3::Criteria` type. All you're doing here is using 3 separate states (room ID, search keyword, and include all rooms boolean) to act as a worse version of the `Criteria` struct.

The UI should be responsible for creating the `Criteria` struct and passing it to this request handler background task. That offers two benefits:
1. The UI can easily support more advanced criteria (both now, or in the future) without needing to change the backend.
2. This task doesn't have to concern itself with configuring the search criteria properly.

You can keep the `room_id` for the sake of convenience (for now), since we only support searching either one room or all rooms.

kevinaboos reviewed (changes requested):
Thanks for your effort on this feature! I know it's quite complicated so I appreciate the work.

I like the new design that re-uses a lot of the RoomScreen/Message widgets and functions for the search panel. That's the right way of doing things.

I have one general question about a UI/UX design choice:
There is a lot of code scattered everywhere that seemingly exists only to hide the message search view when a room is deselected. Why? I don't think we should do that. You can make the code a lot simpler by always showing the search view, and making it default to searching only the current room (if one is selected/focused) or all rooms if none are selected. 

I didn't exhaustively test everything yet. Before I do so, kindly ensure you have tested it thoroughly.

Also, please clean up and polish the UI a bit. The spacing of items looks a little bit messy/unprofessional. Keep in mind that we want Robrix to serve as the flagship example of a complex Makepad app, so it's important to spend the extra time to make the UI as perfect & beautiful as we can.

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -979,6 +1004,68 @@ impl Widget for RoomScreen {
+                            let current_first_index = portal_list.first_id();
+                            portal_list.smooth_scroll_to(
+                                cx,
+                                index.saturating_sub(2),
+                                index.saturating_sub(2).abs_diff(current_first_index) as f64 / SMOOTH_SCROLL_TIME,
```
Change to index.saturating_sub(1)

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -55,6 +55,11 @@ const MESSAGE_NOTICE_TEXT_COLOR: Vec3 = Vec3 { x: 0.5, y: 0.5, z: 0.5 };
 
+/// The time required to scroll to the targeted message in seconds.
+/// 
+/// This value cannot be zero.
+const SMOOTH_SCROLL_TIME: f64 = 0.5;
```
Uses NonZeroU32 to ensure SMOOTH_SCROLL_TIME is not zero. However smooth_scroll with speed that is `inf` still works.

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -4217,6 +4304,12 @@ pub enum MessageAction {
+    /// Scroll to a message in the main room timeline with the given event_id.
+    /// Used when navigating from search results to the original message in the timeline.
+    ScrollToMessage {
+        room_id: matrix_sdk::ruma::OwnedRoomId,
+        event_id: matrix_sdk::ruma::OwnedEventId,
```
Fixed

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -340,6 +345,26 @@ live_design! {
                         }
                         text: "<Username not available>"
                     }
+                    jump_button_view = <View> {
```
Fixed

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -4237,13 +4330,29 @@ pub enum MessageAction {
+/// Contains the information needed to jump to a specific message.
+/// Used primarily for search results to allow navigation to the original message location.
+#[derive(Clone, Debug)]
+pub struct JumpOption {
```
Fixed

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -4257,7 +4366,50 @@ impl Widget for Message {
+                if self.view.button(id!(jump_button_view.jump_button)).clicked(actions) {
+                    if jump_option.from_all_rooms_search {
+                        if let Some(selected_room) = {
+                            let app_state = scope.data.get::<AppState>().unwrap();
+                            app_state.selected_room.clone()
+                        } {
+                            // If room_id is not the selected room, select the room and open its dock tab
+                            if selected_room.room_id() != &jump_option.room_id {
+                                let room_name: Option<String> = {
+                                    let rooms_list_ref = cx.get_global::<RoomsListRef>();
+                                    rooms_list_ref.get_room_name(&jump_option.room_id)
+                                };
+                                
+                                let target_selected_room = SelectedRoom::JoinedRoom {
+                                    room_id: jump_option.room_id.clone().into(),
+                                    room_name,
+                                };
+                                
+                                // Dispatch action to select the room and open its dock tab
+                                cx.widget_action(
+                                    self.widget_uid(),
+                                    &scope.path,
+                                    RoomsListAction::Selected(target_selected_room)
+                                );
+                            }
+                        }
+                    }
+                    
+                    // Always scroll to the message after room selection (if needed)
+                    cx.widget_action(
+                        self.widget_uid(),
+                        &scope.path,
+                        MessageAction::ScrollToMessage {
+                            room_id: jump_option.room_id.clone(),
+                            event_id: jump_option.event_id.clone(),
+                        }
+                    );
+                }
+            }
+        }
```
There is an early return if there is no detail in the message card which search message card does not have. L4512 is after the early return.

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+    use crate::home::room_screen::*;
+
+    COLOR_BUTTON_GREY = #B6BABF
+    ICON_SEARCH = dep("crate://self/resources/icons/search.svg")
```
Fixed

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+            }
+        }
+    }
+    pub MessageCard = <Message> {
```
Fixed

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+    }
+}
+
+/// Yellow color.
+const SEARCH_HIGHLIGHT: Vec3 = Vec3 {
+    x: 1.0,
+    y: 0.87,
+    z: 0.127,
+};
```
Fixed, using COLOR_WARNING_YELLOW instead.

alanpoon committed:
search_message cleanup

alanpoon commented on src/shared/message_search_input_bar.rs:
```diff
@@ -0,0 +1,175 @@
+                return;
+            }
+            clear_button.set_visible(cx, !keywords.is_empty());
+            self.debounce_timer = cx.start_timeout(1.0);
```
Fixed

alanpoon commented on src/sliding_sync.rs:
```diff
@@ -8,53 +8,47 @@ use imbl::Vector;
+    }, utils::{self, AVATAR_THUMBNAIL_FORMAT}, verification::add_verification_event_handlers_and_sync_client
 };
 
+/// Maximum length for search terms to prevent overly long queries
+const MAX_SEARCH_TERM_LENGTH: usize = 500;
```
Removed

alanpoon commented on src/sliding_sync.rs:
```diff
@@ -1147,6 +1155,156 @@ async fn async_worker(
+                    continue;
+                }
+                
+                if search_term.len() > MAX_SEARCH_TERM_LENGTH {
+                    log!("Search term too long: '{}' (maximum {} characters)", search_term, MAX_SEARCH_TERM_LENGTH);
+                    enqueue_popup_notification(PopupItem {
+                        message: format!("Search term cannot exceed {} characters", MAX_SEARCH_TERM_LENGTH),
+                        auto_dismissal_duration: Some(3.0),
+                        kind: PopupKind::Warning,
+                    });
+                    continue;
+                }
```
removed

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+            max_pull_down: 0.0, // set to `0.0` to disable the pulldown bounce animation.
+
+            // Below, we must place all of the possible templates (views) that can be used in the portal list.
+            Message = <Message> {}
```
Removed

alanpoon committed:
minor cleanup

alanpoon committed:
Text improvement

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
change function name

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
search_all_rooms_button fix query twice

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
fix query twice

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
Rephrasing

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
Use 2 dots

Co-authored-by: Kevin Boos <1139460+kevinaboos@users.noreply.github.com>

alanpoon committed:
add new line

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -4324,6 +4476,17 @@ impl Widget for Message {
                         }
                     );
                 }
+                // Event Hits is not captured when Message is in a stack navigation. Hence, cannot implement jump when clicking the message.
```
Fixed

alanpoon commented on src/home/room_screen.rs:
```diff
@@ -4379,11 +4542,37 @@ impl Message {
+        self.view.view(id!(jump_button_view))
+            .set_visible(cx, true);
+        
+        // Event Hit is not captured when Message is in a stack navigation. Hence, cannot implement jump when clicking the message.
```
Fixed

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+        );
+        self.view.view(id!(loading_view)).set_visible(cx, false);
+    }
+    /// Sets the search criteria for the SearchResultSummary widget.
```
fixed

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+    // Use precomputed formatted content
+    let (item, used_cached_item) = if let Some(content) = formatted_content.as_ref() {
+        match &content.msgtype {
+            MessageType::Text(TextMessageEventContent {
+                body, formatted, ..
```
Added.

alanpoon committed:
Added regex

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
Added remove_timeline_loaded_notify

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+                },
+                color: (COLOR_TEXT),
+            }
+            text: "No More"
```
Manually fix

alanpoon commented on src/right_panel/search_message.rs:
```diff
@@ -0,0 +1,1103 @@
+            ))
+        };
+
+        self.view.markdown(id!(summary_label)).set_text(
```
removed markdown

alanpoon commented on src/shared/message_search_input_bar.rs:
```diff
@@ -0,0 +1,175 @@
+    #[rust] search_term: String,
+}
+
+/// Actions emitted by the `MessageSearchInputBar` based on user interaction with it.
```
Removed handling MessageSearchAction actions in search input bar.

alanpoon commented on src/shared/message_search_input_bar.rs:
```diff
@@ -0,0 +1,175 @@
+        }
+        self.view.handle_event(cx, event, scope);
+        let area = self.text_input(id!(input)).area();
+        if let Hit::FingerDown(..) = event.hits(cx, area) {
```
Moved "self.view.handle_event(cx, event, scope);", below event.hits

alanpoon committed:
search message improvement

alanpoon committed:
fix clippy

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
remove notify

alanpoon committed:
Merge branch 'main' into search_messages_3#122

alanpoon committed:
widget action
